download_models/downloadOVMSModels.sh

```diff
@@ -202,27 +203,36 @@ fi
 
 isModelDownloaded() {
     modelName=$1
+    precision=$2
     for m in "$modelDir"/* ; do
         if [ "$(basename "$m")" = "$modelName" ]
         then
-            echo "downloaded"
-            return 0
+            for precision_folder in "$modelDir"/"$modelName"/* ; do
+                if [ "$(basename "$precision_folder")" = "$precision" ]
+                then
+                    echo "downloaded"
+                    return 0
+                fi
+            done
         fi
     done
     echo "not_found"
 }
 
 configFile="$modelDir"/config_template.json
-mapfile -t modelNames < <(docker run -i --rm -v ./:/app ghcr.io/jqlang/jq -r '.model_config_list.[].config.name' < "$configFile")
+mapfile -t model_base_path < <(docker run -i --rm -v ./:/app ghcr.io/jqlang/jq -r '.model_config_list.[].config.base_path' < "$configFile")
 
-for eachModel in "${modelNames[@]}" ; do
-    ret=$(isModelDownloaded "$eachModel")
+for eachModelBasePath in "${model_base_path[@]}" ; do
+    eachModel=$(echo "$eachModelBasePath" | rev | cut -d '/' -f 2 | rev)
+    precision=$(echo "$eachModelBasePath" | rev | cut -d '/' -f 1 | rev)
```

Comment:
suggest using awk is simpler and more performant:
```suggestion
    eachModel=$(echo "$eachModelBasePath" | awk -F/ '{print $(NF-1)}')
    precision=$(echo "$eachModelBasePath" | awk -F/ '{print $NF}')
```